engine/time/src/main/java/io/deephaven/time/calendar/Calendars.java

```diff
@@ -3,109 +3,49 @@
 import io.deephaven.internal.log.LoggerFactory;
+import io.deephaven.io.logger.Logger;
 import org.jetbrains.annotations.NotNull;
 
-import java.io.BufferedReader;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
+import java.io.*;
 import java.nio.file.NoSuchFileException;
-import java.util.Collection;
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Set;
 import java.util.function.Consumer;
 
 /**
  * A collection of business calendars.
  */
-public class Calendars implements Map<String, BusinessCalendar> {
+public class Calendars {
 
     private static final Logger logger = LoggerFactory.getLogger(Calendars.class);
     private static final String BUSINESS_CALENDAR_PROP_INTERNAL = "Calendar.importPath";
     private static final String BUSINESS_CALENDAR_PROP_USER = "Calendar.userImportPath";
-    private static final Calendars instance = new Calendars();
-    private static final String defaultName = Configuration.getInstance().getProperty("Calendar.default");
-
-    /**
-     * Gets the singleton map of business calendars.
-     *
-     * @return singleton map of calendars
-     */
-    static Calendars getInstance() {
-        return instance;
-    }
-
-    /**
-     * Returns a business calendar.
-     *
-     * @param name name of the calendar
-     * @return business calendar
-     * @throws IllegalArgumentException no calendar matching {@code name}
-     */
-    public static BusinessCalendar calendar(final String name) {
-        Require.neqNull(name, "name");
+    private static String defaultName = Configuration.getInstance().getProperty("Calendar.default");
 
-        final String n = name.toUpperCase();
+    private static final Map<String, BusinessCalendar> map = new HashMap<>();
+    private static String[] names = new String[0];
 
-        if (!instance.containsKey(n)) {
-            throw new IllegalArgumentException("No such calendar: " + name);
-        }
+    private Calendars() {}
 
-        return instance.get(n);
-    }
+    // region Load
 
-    /**
-     * Returns a business calendar.
-     *
-     * @return default business calendar. The deault is specified by the {@code Calendar.default} property.
-     */
-    public static BusinessCalendar calendar() {
-        return calendar(defaultName);
-    }
-
-    /**
-     * Returns the default calendar name
-     *
-     * @return default business calendar name
-     */
-    public static String getDefaultName() {
-        return defaultName;
-    }
-
-    /**
-     * Returns the names of all available calendars
-     *
-     * @return names of all available calendars
-     */
-    public static String[] calendarNames() {
-        return instance.keySet().toArray(String[]::new);
-    }
-
-
-
-    private final Map<String, BusinessCalendar> calendars = new HashMap<>();
-
-
-    private Calendars() {
+    static {
```

Comment:
Deferring complex work that can result in errors after class initialization is helpful for error reporting.  This is going to end up with hard to diagnose class not found errors.

Reply:
Would you prefer all methods to have lazy init class as opposed to the static init?

Reply:
Deferring.